src/pages/FormPageHoc.tsx

```diff
@@ -0,0 +1,59 @@
+import { useState } from "react";
+import { Link } from "react-router-dom";
+
+import { ValidationErrorItem } from "joi";
+
+import { Button } from "@lib/gds-components";
+import {
+  DevelopmentPlan,
+  DevelopmentPlanTimetable,
+} from "@lib/types/timetable";
+import { useFormContext } from "../context/use-form-context";
+import { useSequence } from "./use-sequence";
+
+import styles from "./form-page.module.css";
+
+export const FormPageHoC = <P extends Record<string, unknown>>(
+  FormComponent: (
+    props: P & { errors: ValidationErrorItem[] | undefined }
+  ) => JSX.Element,
+  formProps: P,
+  validatePage?: (
+    developmentPlan: DevelopmentPlan,
+    developmentPlanTimetable: DevelopmentPlanTimetable[],
+    formProps: P
+  ) => ValidationErrorItem[]
+) => {
+  const InnerComponent = () => {
```

Comment:
Is defining a component within a component okay?

Reply:
Yeah, that's fine. This is a higher order component, essentially the point is to add some extra functionality around the component you want to render (in this case, navigation and validation handling)